interface/patient_file/history/history.php

```diff
@@ -76,24 +81,43 @@
                 $result = getHistoryData($pid);
             }
             ?>
-
-            <?php if (acl_check('patients', 'med', '', array('write','addonly'))) { ?>
-                <div class="page-header">
-                    <h2><?php echo htmlspecialchars(getPatientName($pid), ENT_NOQUOTES);?> <small><?php echo xl("History & Lifestyle");?></small></h2>
-                </div>
-                <div>
-                    <div class="btn-group">
-                        <a href="../summary/demographics.php" class="btn btn-default btn-back" onclick="top.restoreSession()">
-                            <?php echo htmlspecialchars(xl('Back To Patient'), ENT_NOQUOTES);?>
-                        </a>
-                        <a href="history_full.php" class="btn btn-default btn-edit" onclick="top.restoreSession()">
-                            <?php echo htmlspecialchars(xl("Edit"), ENT_NOQUOTES);?>
-                        </a>
-                    </div>
+        </div>
+    </div>
+    <?php
+    if (acl_check('patients', 'med', '', array('write','addonly'))) {
+        $header_title = xl('History and Lifestyle of');?>
+        <div class="row">
+            <div class="col-sm-12">
+                <?php
+                   $portal_login_href = "../summary/create_portallogin.php";
+                   require_once("../summary/dashboard_header.php");
+                ?>
+            </div>
+        </div>
+        <div class="row" >
+            <div class="col-sm-12">
+                <?php
+                    $list_id = "nav-list2"; // to indicate nav item is active, count and give correct id
+                    // Collect the patient menu then build it
+                    $menuPatient = new PatientMenuRole();
+                    $menu_restrictions = $menuPatient->getMenu();
+                    require_once("../summary/dashboard_nav.php");
```

Comment:
We could make this mechanism much more sleek if did:
```
// Display the patient menu
$menuPatient = new PatientMenuRole();
$menuPatient->displayMenu();
```

And then add the following function in the PatientMenuRole class:
```
public function displayMenu()
{
$menu_restrictions = $this->getMenu();
// then just echo everything in the dashboard_nav.php script
}
```
(and then you can delete the dashboard_nav.php script)

Reply:
I didn't want to bring bootstrap code into that class :)

Reply:
I wouldn't worry about that. Modularizing it in a class function will make it easier for folks to:
1. add it to their customized pages (ie. which folks will want to do since the menu itself is highly customizable)
2. customize/expand it as needed in the future via function parameters

Reply:
Couple of points:
This menu works well in this situation, it also needs some additional jQuery in the page to mimic a tabbed interface. 
By keeping the `dashboard_nav.php` script out you get an array of  objects containing the absolute url that can be used in any way to create different menus, you wouldn't be tied to the navbar code
Also I wonder how easy or difficult it will be at a later date when the move to later versions of Bootstrap is made

Reply:
To clarify, I am recommended a new function in the class to output the navbar (ie. the getMenu() function does not get touched, which is what produced the array of objects). I'll leave this decision up to you. Other things to think about (in addition to stuff I already posted above):
1. 1 less script will exist
2. avoid security issues by having another script around that can be run without authentication (note it is fine now, but you never know what somebody may do to it in the future for customization etc).